src/sdk/Player.ts

```diff
@@ -438,21 +452,10 @@ export class Player extends Unit {
           });
         });
         // Create paths to all npc tiles
-        const potentialPaths = map(seekingTiles, (point) =>
-          Pathing.constructPath(this.region, this.location, { x: point.x, y: point.y }),
-        );
-        const potentialPathLengths = map(potentialPaths, (path) => path.length);
-        // Figure out what the min distance is
-        const shortestPathLength = min(potentialPathLengths);
-        // Get all of the paths of the same minimum distance (can be more than 1)
-        const shortestPaths = filter(
-          map(potentialPathLengths, (length, index) => (length === shortestPathLength ? seekingTiles[index] : null)),
-        );
-        // Take the path that is the shortest absolute distance from player
-        this.destinationLocation = minBy(shortestPaths, (point) =>
-          Pathing.dist(this.location.x, this.location.y, point.x, point.y),
-        );
+        const path = Pathing.constructPaths(this.region, this.location, seekingTiles);
```

Comment:
the meat of the optimisation is the ability to call this once instead of dozens of times on zuk (where there is no path at all)